modules/product/src/main/java/com/opengamma/strata/product/Attributes.java

```diff
@@ -61,6 +61,17 @@ public static Attributes empty() {
         Messages.format("Attribute not found for type '{}'", type)));
   }
 
+  /**
+   * Determines if an attribute associated with the specified type is present.
+   *
+   * @param <T>  the type of the attribute value
+   * @param type  the type to find
+   * @return true if a matching attribute is present
+   */
+  public default <T> boolean hasAttribute(AttributeType<T> type) {
```

Comment:
Looking around, we tend to use `containsXxx()` for methods like this.

Reply:
Updated.